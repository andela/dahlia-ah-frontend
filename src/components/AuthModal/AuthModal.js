/* eslint-disable jsx-a11y/control-has-associated-label */
/* eslint-disable jsx-a11y/interactive-supports-focus */

import React from 'react';
import PropType from 'prop-types';
import './AuthModal.scss';

const AuthModal = ({
  children, title, desc, isOpen, closeModal,
}) => (
  <div className="backdrop-div" style={{ display: isOpen ? 'flex' : 'none' }}>
    <div className="modal">
      <div className="modal-content">
        <i className="fas fa-times" role="button" onKeyDown={closeModal} onClick={closeModal} />
        <h4>{title}</h4>
        <p className="description">{desc}</p>
        {children}
      </div>
    </div>
  </div>
);

AuthModal.propTypes = {
  children: PropType.node.isRequired,
  title: PropType.string.isRequired,
  desc: PropType.string.isRequired,
  isOpen: PropType.bool.isRequired,
  closeModal: PropType.func.isRequired,
};

export default AuthModal;
