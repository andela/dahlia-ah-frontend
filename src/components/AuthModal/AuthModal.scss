.backdrop-div {
  display: flex;
  align-items: center;
  position: absolute;
  width: 100%;
  height: 100vh;
  background: rgba(0, 0, 0, 0.6);

  h4 {
      font-size: 24px;
      line-height: 28px;
      font-weight: 500;
  }

  .description {
      font-style: normal;
      font-weight: normal;
      font-size: 18px;
      line-height: 21px;
      text-align: center;
      color: #747474;
  }
  .modal-content {
    padding: 50px 70px;
    .fa-times {
        position: absolute;
        top:15px;
        right:15px;
        font-size: 18px;
        cursor: pointer;
    }
    > div {
      margin: 30px 0;
  }
  }
}
.modal {
  display: block;
  text-align: center;
  max-width: 487px;
  min-height: 350px;
  max-height: 750px;
}

@media screen and (max-width: 500px) and (min-width: 320px) {
  .backdrop-div {
    h4 {
        font-size: 18px;
    }
  
    .description {
        font-size: 14px;
    }
}
.modal {
  min-height: 280px;
}
}

@media screen and (max-width: 330px) {
  .backdrop-div {

    .description {
        font-size: 12px;
    }
  }
}

@media screen and (max-width: 420px) and (min-width: 300px) {
  .backdrop-div {
    .modal-content {
      padding: 30px 30px;
    }
  }
}
