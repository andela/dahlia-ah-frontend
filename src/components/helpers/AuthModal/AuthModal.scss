.backdrop-div {
    display: flex;
    position: fixed;
    align-items: center;
    z-index: 1000;
    width: 100%;
    height: 100%;
    background: rgba(0, 0, 0, 0.6);

  
    h4 {
        font-size: 24px;
        line-height: 28px;
        font-weight: 500;
        color: #000000
    }
  
    .description {
        font-style: normal;
        font-weight: normal;
        font-size: 18px;
        line-height: 21px;
        text-align: center;
        color: #747474;
    }
  
    .modal-content {
        padding: 40px 70px 10px 70px;
  
        .fa-times {
            position: absolute;
            top:15px;
            right:15px;
            font-size: 18px;
            cursor: pointer;
            color: #000000;
        }
        > div {
          margin: 30px 0;
        }
    }
  
  .modal {
    display: block;
    text-align: center;
    max-width: 487px;
    min-height: 350px;
    max-height: 750px;
    padding: 0;
  }
  
  @media screen and (max-width: 500px) and (min-width: 320px) {
    h4 {
      font-size: 18px;
    }
    .description {
      font-size: 14px;
    }
    .modal {
      min-height: 280px;
      max-height: 560px;
    }
  }
  
  @media screen and (max-width: 330px) {
        .description {
            font-size: 12px;
        }
  }
  
  @media screen and (max-width: 420px) and (min-width: 300px) {
      .modal-content {
        padding: 10px 30px;
      }
  }
}