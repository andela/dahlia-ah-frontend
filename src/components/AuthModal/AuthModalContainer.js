import React, { useState } from 'react';
import PropTypes from 'prop-types';
import AuthModal from './AuthModal';
import './AuthModal.scss';

const AuthModalContainer = ({
  children, title, desc, isOpen,
}) => {
  const [display, setDisplay] = useState(isOpen);
  const closeModal = () => {
    setDisplay(false);
  };

  return (
    <AuthModal title={title} desc={desc} isOpen={isOpen} display={display} closeModal={closeModal}>
      {children}
    </AuthModal>
  );
};

AuthModalContainer.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  title: PropTypes.string.isRequired,
  desc: PropTypes.string.isRequired,
  isOpen: PropTypes.bool.isRequired,
};

export default AuthModalContainer;
