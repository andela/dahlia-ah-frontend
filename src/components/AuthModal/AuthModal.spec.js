import React from 'react';
import { configure, shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import AuthModal from './AuthModal';

configure({ adapter: new Adapter() });

describe('AuthModal component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <AuthModal title="WELCOME BACK" desc="Sign in using your social account" isOpen closeModal={() => { }}>
        <div>Test</div>
      </AuthModal>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
