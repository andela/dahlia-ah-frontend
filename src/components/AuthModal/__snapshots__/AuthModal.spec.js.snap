// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthModal component renders correctly 1`] = `ShallowWrapper {}`;
